### DISCUSSION TITLE
Stability checks incorrectly pass when force retrying/rebasing a branch

### DISCUSSION THREAD

**Discussion by andrewgies17** [renovatebot/renovate#22189] [Suggest an Idea]
Discovered this while trying to create a reproduction of https://github.com/renovatebot/renovate/issues/5743 in a repo that I can share publicly - no such luck, but this is something...

**What Renovate type, platform and version are you using?**

Using self-hosted renovate targeting GitLab. Right now, running 23.3.1 locally (npm install -g renovate).
Minimal reproduction here: https://gitlab.com/andrew.gies.axiom/renovate-stability-repro
Incorrect checks visible in the latest pipeline run on this MR: https://gitlab.com/andrew.gies.axiom/renovate-stability-repro/-/merge_requests/1/pipelines

**Describe the bug**

The stability-days status check sometimes sets an incorrect value when run. For example, the stability days option changed from "not stable" (the correct value) to "stable", then back again across different runs of renovate. I believe what's happening is whenever the "force retry/rebase" checkbox in the master issue is used to re-create an MR, the stability checks aren't properly being updated (just being set to green by default). However, when you re-run renovate without that checkbox, it'll go back and fix the status check.

**Relevant debug logs**

The full logs for my three runs are here: https://gitlab.com/andrew.gies.axiom/renovate-stability-repro/-/snippets/2011587

The first run was without the checkbox, and created the MR and the first pipeline. The second run was with the force-retry, and created the second pipeline with the incorrect passing check. The third run (normal, no checkbox) revised the passing check to be not-passing (correctly) without creating a new pipeline. A fourth run (not included in logs), was run with the checkbox to create a third pipeline with the incorrect check, just so I can include it as an example of the failure in this issue.

I've identified a handful of key locations within these logs where the issue is happening, those line numbers are listed in the description of the snippet.

**To Reproduce**

- Create a repo with a renovate configuration that uses stability days and the master issue.
- Add an npm dependency that should fail the stability check.
- Run renovate and see it creates an MR with the failing check, correctly.
- Check the "force retry" box in the master issue and run renovate again.
- Notice that the check is now passing, incorrectly.

**Comment by viceice** [comment]
Thanks for reproduction. 

I think the current renovate behavior is by design but can be a bug too.

So you discovered, that renovate will always set  stability days to green when force rebase was checked?

Did you use the pr or the dashboard issue checkbox?

**Comment by viceice** [comment]
I'm assuming that this behavior is identical on our other platforms too.

**Comment by andrewgies17** [comment]
This was reproduced using the issue dashboard checkbox, I haven't tried using the PR checkbox.
